torch/csrc/jit/script/compiler.cpp

```diff
@@ -508,6 +462,39 @@ inline bool isSupportedListElementType(const TypePtr& type) {
       type->isSubtypeOf(NumberType::get());
 }
 
+using MiniTypeEnvironment = MiniEnvironment<TypePtr>;
+
+// Invokes close_var_fn when a variable is used that isn't defined in the
```

Comment:
I think this is code smell due to an ordering issue. I think lambda lifting forks and creating closures should happen _after_ we convert to SSA. They benefit a lot from being able to see how a use links to a definition, which is exactly what the SSA conversion provides. It is likely the ideal order is something like:

* emit
* convertSSA
* liftFork/liftClosure
* inline closure calls.